{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "497b4641",
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5",
    "execution": {
     "iopub.execute_input": "2024-12-04T10:57:29.574735Z",
     "iopub.status.busy": "2024-12-04T10:57:29.574046Z",
     "iopub.status.idle": "2024-12-04T10:58:13.396640Z",
     "shell.execute_reply": "2024-12-04T10:58:13.395488Z"
    },
    "papermill": {
     "duration": 43.828078,
     "end_time": "2024-12-04T10:58:13.398740",
     "exception": false,
     "start_time": "2024-12-04T10:57:29.570662",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting ray==2.10.0\r\n",
      "  Downloading ray-2.10.0-cp310-cp310-manylinux2014_x86_64.whl.metadata (13 kB)\r\n",
      "Requirement already satisfied: click>=7.0 in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (8.1.7)\r\n",
      "Requirement already satisfied: filelock in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (3.15.1)\r\n",
      "Requirement already satisfied: jsonschema in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (4.22.0)\r\n",
      "Requirement already satisfied: msgpack<2.0.0,>=1.0.0 in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (1.0.8)\r\n",
      "Requirement already satisfied: packaging in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (21.3)\r\n",
      "Requirement already satisfied: protobuf!=3.19.5,>=3.15.3 in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (3.20.3)\r\n",
      "Requirement already satisfied: pyyaml in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (6.0.2)\r\n",
      "Requirement already satisfied: aiosignal in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (1.3.1)\r\n",
      "Requirement already satisfied: frozenlist in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (1.4.1)\r\n",
      "Requirement already satisfied: requests in /opt/conda/lib/python3.10/site-packages (from ray==2.10.0) (2.32.3)\r\n",
      "Requirement already satisfied: attrs>=22.2.0 in /opt/conda/lib/python3.10/site-packages (from jsonschema->ray==2.10.0) (23.2.0)\r\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /opt/conda/lib/python3.10/site-packages (from jsonschema->ray==2.10.0) (2023.12.1)\r\n",
      "Requirement already satisfied: referencing>=0.28.4 in /opt/conda/lib/python3.10/site-packages (from jsonschema->ray==2.10.0) (0.35.1)\r\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in /opt/conda/lib/python3.10/site-packages (from jsonschema->ray==2.10.0) (0.18.1)\r\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /opt/conda/lib/python3.10/site-packages (from packaging->ray==2.10.0) (3.1.2)\r\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests->ray==2.10.0) (3.3.2)\r\n",
      "Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests->ray==2.10.0) (3.7)\r\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/conda/lib/python3.10/site-packages (from requests->ray==2.10.0) (1.26.18)\r\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests->ray==2.10.0) (2024.6.2)\r\n",
      "Downloading ray-2.10.0-cp310-cp310-manylinux2014_x86_64.whl (65.1 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m65.1/65.1 MB\u001b[0m \u001b[31m27.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: ray\r\n",
      "  Attempting uninstall: ray\r\n",
      "    Found existing installation: ray 2.24.0\r\n",
      "    Uninstalling ray-2.24.0:\r\n",
      "      Successfully uninstalled ray-2.24.0\r\n",
      "Successfully installed ray-2.10.0\r\n",
      "Collecting autogluon.tabular\r\n",
      "  Downloading autogluon.tabular-1.2-py3-none-any.whl.metadata (14 kB)\r\n",
      "Requirement already satisfied: numpy<2.1.4,>=1.25.0 in /opt/conda/lib/python3.10/site-packages (from autogluon.tabular) (1.26.4)\r\n",
      "Requirement already satisfied: scipy<1.16,>=1.5.4 in /opt/conda/lib/python3.10/site-packages (from autogluon.tabular) (1.14.1)\r\n",
      "Requirement already satisfied: pandas<2.3.0,>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from autogluon.tabular) (2.2.3)\r\n",
      "Collecting scikit-learn<1.5.3,>=1.4.0 (from autogluon.tabular)\r\n",
      "  Downloading scikit_learn-1.5.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\r\n",
      "Requirement already satisfied: networkx<4,>=3.0 in /opt/conda/lib/python3.10/site-packages (from autogluon.tabular) (3.3)\r\n",
      "Collecting autogluon.core==1.2 (from autogluon.tabular)\r\n",
      "  Downloading autogluon.core-1.2-py3-none-any.whl.metadata (12 kB)\r\n",
      "Collecting autogluon.features==1.2 (from autogluon.tabular)\r\n",
      "  Downloading autogluon.features-1.2-py3-none-any.whl.metadata (11 kB)\r\n",
      "Requirement already satisfied: tqdm<5,>=4.38 in /opt/conda/lib/python3.10/site-packages (from autogluon.core==1.2->autogluon.tabular) (4.66.4)\r\n",
      "Requirement already satisfied: requests in /opt/conda/lib/python3.10/site-packages (from autogluon.core==1.2->autogluon.tabular) (2.32.3)\r\n",
      "Requirement already satisfied: matplotlib<3.11,>=3.7.0 in /opt/conda/lib/python3.10/site-packages (from autogluon.core==1.2->autogluon.tabular) (3.7.5)\r\n",
      "Requirement already satisfied: boto3<2,>=1.10 in /opt/conda/lib/python3.10/site-packages (from autogluon.core==1.2->autogluon.tabular) (1.26.100)\r\n",
      "Collecting autogluon.common==1.2 (from autogluon.core==1.2->autogluon.tabular)\r\n",
      "  Downloading autogluon.common-1.2-py3-none-any.whl.metadata (11 kB)\r\n",
      "Requirement already satisfied: psutil<7.0.0,>=5.7.3 in /opt/conda/lib/python3.10/site-packages (from autogluon.common==1.2->autogluon.core==1.2->autogluon.tabular) (5.9.3)\r\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /opt/conda/lib/python3.10/site-packages (from pandas<2.3.0,>=2.0.0->autogluon.tabular) (2.9.0.post0)\r\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/conda/lib/python3.10/site-packages (from pandas<2.3.0,>=2.0.0->autogluon.tabular) (2024.1)\r\n",
      "Requirement already satisfied: tzdata>=2022.7 in /opt/conda/lib/python3.10/site-packages (from pandas<2.3.0,>=2.0.0->autogluon.tabular) (2024.1)\r\n",
      "Requirement already satisfied: joblib>=1.2.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn<1.5.3,>=1.4.0->autogluon.tabular) (1.4.2)\r\n",
      "Requirement already satisfied: threadpoolctl>=3.1.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn<1.5.3,>=1.4.0->autogluon.tabular) (3.5.0)\r\n",
      "Collecting botocore<1.30.0,>=1.29.100 (from boto3<2,>=1.10->autogluon.core==1.2->autogluon.tabular)\r\n",
      "  Downloading botocore-1.29.165-py3-none-any.whl.metadata (5.9 kB)\r\n",
      "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /opt/conda/lib/python3.10/site-packages (from boto3<2,>=1.10->autogluon.core==1.2->autogluon.tabular) (1.0.1)\r\n",
      "Requirement already satisfied: s3transfer<0.7.0,>=0.6.0 in /opt/conda/lib/python3.10/site-packages (from boto3<2,>=1.10->autogluon.core==1.2->autogluon.tabular) (0.6.2)\r\n",
      "Requirement already satisfied: contourpy>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (1.2.1)\r\n",
      "Requirement already satisfied: cycler>=0.10 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (0.12.1)\r\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (4.53.0)\r\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (1.4.5)\r\n",
      "Requirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (21.3)\r\n",
      "Requirement already satisfied: pillow>=6.2.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (10.3.0)\r\n",
      "Requirement already satisfied: pyparsing>=2.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib<3.11,>=3.7.0->autogluon.core==1.2->autogluon.tabular) (3.1.2)\r\n",
      "Requirement already satisfied: six>=1.5 in /opt/conda/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas<2.3.0,>=2.0.0->autogluon.tabular) (1.16.0)\r\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests->autogluon.core==1.2->autogluon.tabular) (3.3.2)\r\n",
      "Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests->autogluon.core==1.2->autogluon.tabular) (3.7)\r\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/conda/lib/python3.10/site-packages (from requests->autogluon.core==1.2->autogluon.tabular) (1.26.18)\r\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests->autogluon.core==1.2->autogluon.tabular) (2024.6.2)\r\n",
      "Downloading autogluon.tabular-1.2-py3-none-any.whl (352 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m352.2/352.2 kB\u001b[0m \u001b[31m15.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading autogluon.core-1.2-py3-none-any.whl (266 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m266.2/266.2 kB\u001b[0m \u001b[31m17.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading autogluon.features-1.2-py3-none-any.whl (64 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.1/64.1 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading autogluon.common-1.2-py3-none-any.whl (68 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m68.2/68.2 kB\u001b[0m \u001b[31m4.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading scikit_learn-1.5.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.3 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m90.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading botocore-1.29.165-py3-none-any.whl (11.0 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m11.0/11.0 MB\u001b[0m \u001b[31m105.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: scikit-learn, botocore, autogluon.common, autogluon.features, autogluon.core, autogluon.tabular\r\n",
      "  Attempting uninstall: scikit-learn\r\n",
      "    Found existing installation: scikit-learn 1.2.2\r\n",
      "    Uninstalling scikit-learn-1.2.2:\r\n",
      "      Successfully uninstalled scikit-learn-1.2.2\r\n",
      "  Attempting uninstall: botocore\r\n",
      "    Found existing installation: botocore 1.35.36\r\n",
      "    Uninstalling botocore-1.35.36:\r\n",
      "      Successfully uninstalled botocore-1.35.36\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "aiobotocore 2.15.2 requires botocore<1.35.37,>=1.35.16, but you have botocore 1.29.165 which is incompatible.\r\n",
      "bigframes 0.22.0 requires google-cloud-bigquery[bqstorage,pandas]>=3.10.0, but you have google-cloud-bigquery 2.34.4 which is incompatible.\r\n",
      "bigframes 0.22.0 requires google-cloud-storage>=2.0.0, but you have google-cloud-storage 1.44.0 which is incompatible.\r\n",
      "bigframes 0.22.0 requires pandas<2.1.4,>=1.5.0, but you have pandas 2.2.3 which is incompatible.\r\n",
      "cesium 0.12.3 requires numpy<3.0,>=2.0, but you have numpy 1.26.4 which is incompatible.\r\n",
      "dataproc-jupyter-plugin 0.1.79 requires pydantic~=1.10.0, but you have pydantic 2.10.1 which is incompatible.\r\n",
      "s3fs 2024.9.0 requires fsspec==2024.9.0.*, but you have fsspec 2024.6.0 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed autogluon.common-1.2 autogluon.core-1.2 autogluon.features-1.2 autogluon.tabular-1.2 botocore-1.29.165 scikit-learn-1.5.2\r\n",
      "Requirement already satisfied: ipywidgets in /opt/conda/lib/python3.10/site-packages (8.1.5)\r\n",
      "Requirement already satisfied: comm>=0.1.3 in /opt/conda/lib/python3.10/site-packages (from ipywidgets) (0.2.2)\r\n",
      "Requirement already satisfied: ipython>=6.1.0 in /opt/conda/lib/python3.10/site-packages (from ipywidgets) (8.21.0)\r\n",
      "Requirement already satisfied: traitlets>=4.3.1 in /opt/conda/lib/python3.10/site-packages (from ipywidgets) (5.14.3)\r\n",
      "Requirement already satisfied: widgetsnbextension~=4.0.12 in /opt/conda/lib/python3.10/site-packages (from ipywidgets) (4.0.13)\r\n",
      "Requirement already satisfied: jupyterlab-widgets~=3.0.12 in /opt/conda/lib/python3.10/site-packages (from ipywidgets) (3.0.13)\r\n",
      "Requirement already satisfied: decorator in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (5.1.1)\r\n",
      "Requirement already satisfied: jedi>=0.16 in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (0.19.1)\r\n",
      "Requirement already satisfied: matplotlib-inline in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (0.1.7)\r\n",
      "Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.41 in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (3.0.47)\r\n",
      "Requirement already satisfied: pygments>=2.4.0 in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (2.18.0)\r\n",
      "Requirement already satisfied: stack-data in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (0.6.2)\r\n",
      "Requirement already satisfied: exceptiongroup in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (1.2.0)\r\n",
      "Requirement already satisfied: pexpect>4.3 in /opt/conda/lib/python3.10/site-packages (from ipython>=6.1.0->ipywidgets) (4.9.0)\r\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.3 in /opt/conda/lib/python3.10/site-packages (from jedi>=0.16->ipython>=6.1.0->ipywidgets) (0.8.4)\r\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /opt/conda/lib/python3.10/site-packages (from pexpect>4.3->ipython>=6.1.0->ipywidgets) (0.7.0)\r\n",
      "Requirement already satisfied: wcwidth in /opt/conda/lib/python3.10/site-packages (from prompt-toolkit<3.1.0,>=3.0.41->ipython>=6.1.0->ipywidgets) (0.2.13)\r\n",
      "Requirement already satisfied: executing>=1.2.0 in /opt/conda/lib/python3.10/site-packages (from stack-data->ipython>=6.1.0->ipywidgets) (2.0.1)\r\n",
      "Requirement already satisfied: asttokens>=2.1.0 in /opt/conda/lib/python3.10/site-packages (from stack-data->ipython>=6.1.0->ipywidgets) (2.4.1)\r\n",
      "Requirement already satisfied: pure-eval in /opt/conda/lib/python3.10/site-packages (from stack-data->ipython>=6.1.0->ipywidgets) (0.2.2)\r\n",
      "Requirement already satisfied: six>=1.12.0 in /opt/conda/lib/python3.10/site-packages (from asttokens>=2.1.0->stack-data->ipython>=6.1.0->ipywidgets) (1.16.0)\r\n"
     ]
    }
   ],
   "source": [
    "!pip install ray==2.10.0\n",
    "!pip install autogluon.tabular\n",
    "!pip install -U ipywidgets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "db2e60d5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-12-04T10:58:13.409554Z",
     "iopub.status.busy": "2024-12-04T10:58:13.409211Z",
     "iopub.status.idle": "2024-12-04T10:58:16.185805Z",
     "shell.execute_reply": "2024-12-04T10:58:16.185090Z"
    },
    "papermill": {
     "duration": 2.784308,
     "end_time": "2024-12-04T10:58:16.187942",
     "exception": false,
     "start_time": "2024-12-04T10:58:13.403634",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "from autogluon.tabular import TabularPredictor\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6054bdf4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-12-04T10:58:16.198539Z",
     "iopub.status.busy": "2024-12-04T10:58:16.197918Z",
     "iopub.status.idle": "2024-12-04T10:58:28.805426Z",
     "shell.execute_reply": "2024-12-04T10:58:28.804446Z"
    },
    "papermill": {
     "duration": 12.614785,
     "end_time": "2024-12-04T10:58:28.807438",
     "exception": false,
     "start_time": "2024-12-04T10:58:16.192653",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "train = pd.read_csv(\"/kaggle/input/competition/oof_data.csv\")\n",
    "test = pd.read_csv(\"/kaggle/input/competition/test_data.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6a534762",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-12-04T10:58:28.818841Z",
     "iopub.status.busy": "2024-12-04T10:58:28.818552Z",
     "iopub.status.idle": "2024-12-04T15:59:03.463839Z",
     "shell.execute_reply": "2024-12-04T15:59:03.462812Z"
    },
    "papermill": {
     "duration": 18034.653512,
     "end_time": "2024-12-04T15:59:03.465648",
     "exception": false,
     "start_time": "2024-12-04T10:58:28.812136",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No path specified. Models will be saved in: \"AutogluonModels/ag-20241204_105828\"\n",
      "2024-12-04 10:58:32,065\tINFO worker.py:1752 -- Started a local Ray instance.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_ray_fit pid=495)\u001b[0m 1 warning generated.\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \tWarning: Reducing model 'n_estimators' from 300 -> 143 due to low time. Expected time usage reduced from 5839.1s -> 2786.3s...\n",
      "\u001b[36m(_ray_fit pid=1342)\u001b[0m \tWarning: CatBoost on GPU is experimental. If you encounter issues, use CPU for training CatBoost instead.\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 46, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return pickle.loads(ray_exception.serialized_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m ModuleNotFoundError: No module named '_catboost'\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m The above exception was the direct cause of the following exception:\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 404, in deserialize_objects\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     obj = self._deserialize_object(data, metadata, object_ref)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 293, in _deserialize_object\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_bytes(obj)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 40, in from_bytes\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_ray_exception(ray_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 49, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     raise RuntimeError(msg) from e\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m RuntimeError: Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \tWarning: Exception caused CatBoost_BAG_L1 to fail during training... Skipping this model.\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \t\tSystem error: Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m traceback: Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 46, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return pickle.loads(ray_exception.serialized_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m ModuleNotFoundError: No module named '_catboost'\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m The above exception was the direct cause of the following exception:\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 404, in deserialize_objects\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     obj = self._deserialize_object(data, metadata, object_ref)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 293, in _deserialize_object\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_bytes(obj)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 40, in from_bytes\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_ray_exception(ray_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 49, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     raise RuntimeError(msg) from e\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m RuntimeError: Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Detailed Traceback:\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/trainer/abstract_trainer.py\", line 2106, in _train_and_save\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     model = self._train_single(**model_fit_kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/trainer/abstract_trainer.py\", line 1993, in _train_single\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, X_test=X_test, y_test=y_test, total_resources=total_resources, **model_fit_kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/abstract/abstract_model.py\", line 925, in fit\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     out = self._fit(**kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/stacker_ensemble_model.py\", line 270, in _fit\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/bagged_ensemble_model.py\", line 298, in _fit\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     self._fit_folds(\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/bagged_ensemble_model.py\", line 724, in _fit_folds\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     fold_fitting_strategy.after_all_folds_scheduled()\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/fold_fitting_strategy.py\", line 690, in after_all_folds_scheduled\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     self._run_parallel(X, y, X_pseudo, y_pseudo, model_base_ref, time_limit_fold, head_node_id)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/fold_fitting_strategy.py\", line 631, in _run_parallel\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     self._process_fold_results(finished, unfinished, fold_ctx)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/fold_fitting_strategy.py\", line 587, in _process_fold_results\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     raise processed_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/autogluon/core/models/ensemble/fold_fitting_strategy.py\", line 550, in _process_fold_results\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     fold_model, pred_proba, time_start_fit, time_end_fit, predict_time, predict_1_time, predict_n_size, fit_num_cpus, fit_num_gpus = self.ray.get(finished)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/auto_init_hook.py\", line 21, in auto_init_wrapper\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return fn(*args, **kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/client_mode_hook.py\", line 103, in wrapper\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return func(*args, **kwargs)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/worker.py\", line 2667, in get\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/worker.py\", line 866, in get_objects\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     raise value\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m ray.exceptions.RaySystemError: System error: Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m traceback: Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 46, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return pickle.loads(ray_exception.serialized_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m ModuleNotFoundError: No module named '_catboost'\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m The above exception was the direct cause of the following exception:\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m Traceback (most recent call last):\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 404, in deserialize_objects\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     obj = self._deserialize_object(data, metadata, object_ref)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/_private/serialization.py\", line 293, in _deserialize_object\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_bytes(obj)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 40, in from_bytes\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     return RayError.from_ray_exception(ray_exception)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m   File \"/opt/conda/lib/python3.10/site-packages/ray/exceptions.py\", line 49, in from_ray_exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m     raise RuntimeError(msg) from e\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m RuntimeError: Failed to unpickle serialized exception\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \tWarning: Model is expected to require 294.5s to train, which exceeds the maximum time limit of 13.3s, skipping model...\n",
      "\u001b[36m(_ray_fit pid=1518)\u001b[0m \tWarning: Model has no time left to train, skipping model... (Time Left = -0.2s)\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \tWarning: Model has no time left to train, skipping model... (Time Left = -0.1s)\u001b[32m [repeated 2x across cluster] (Ray deduplicates logs by default. Set RAY_DEDUP_LOGS=0 to disable log deduplication, or see https://docs.ray.io/en/master/ray-observability/ray-logging.html#log-deduplication for more options.)\u001b[0m\n",
      "\u001b[36m(_dystack pid=181)\u001b[0m \tWarning: Reducing model 'n_estimators' from 300 -> 48 due to low time. Expected time usage reduced from 7981.8s -> 1286.1s...\n",
      "\tWarning: Reducing model 'n_estimators' from 300 -> 108 due to low time. Expected time usage reduced from 11748.6s -> 4246.3s...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Summary of fit() ***\n",
      "Estimated performance of each model:\n",
      "                     model  score_val              eval_metric  pred_time_val      fit_time  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order\n",
      "0      WeightedEnsemble_L3  -1.045002  root_mean_squared_error     137.357594  13280.124841                0.015671           1.770248            3       True         14\n",
      "1      WeightedEnsemble_L2  -1.045053  root_mean_squared_error     101.808019   8417.333692                0.015976           1.487336            2       True         10\n",
      "2        LightGBMXT_BAG_L2  -1.045101  root_mean_squared_error     118.528643   9019.171870                7.015970         117.946934            2       True         11\n",
      "3          LightGBM_BAG_L2  -1.045168  root_mean_squared_error     115.705110   9004.995558                4.192436         103.770621            2       True         12\n",
      "4          LightGBM_BAG_L1  -1.045182  root_mean_squared_error       4.918388     92.356826                4.918388          92.356826            1       True          2\n",
      "5           XGBoost_BAG_L1  -1.045183  root_mean_squared_error       1.528545     68.977578                1.528545          68.977578            1       True          7\n",
      "6        LightGBMXT_BAG_L1  -1.045196  root_mean_squared_error      17.215012    154.019336               17.215012         154.019336            1       True          1\n",
      "7          CatBoost_BAG_L1  -1.045244  root_mean_squared_error       0.274791     84.403559                0.274791          84.403559            1       True          4\n",
      "8   NeuralNetFastAI_BAG_L1  -1.045292  root_mean_squared_error       6.995614   1384.999068                6.995614        1384.999068            1       True          6\n",
      "9     ExtraTreesMSE_BAG_L1  -1.046529  root_mean_squared_error      28.833366    335.394234               28.833366         335.394234            1       True          5\n",
      "10    LightGBMLarge_BAG_L1  -1.046633  root_mean_squared_error       3.977910     82.766158                3.977910          82.766158            1       True          9\n",
      "11  RandomForestMSE_BAG_L1  -1.047952  root_mean_squared_error      42.301118   6380.099314               42.301118        6380.099314            1       True          3\n",
      "12  RandomForestMSE_BAG_L2  -1.049720  root_mean_squared_error     130.325953  13160.407660               18.813280        4259.182723            2       True         13\n",
      "13   NeuralNetTorch_BAG_L1  -1.072592  root_mean_squared_error       5.467930    318.208864                5.467930         318.208864            1       True          8\n",
      "Number of models trained: 14\n",
      "Types of models trained:\n",
      "{'StackerEnsembleModel_TabularNeuralNetTorch', 'StackerEnsembleModel_LGB', 'StackerEnsembleModel_NNFastAiTabular', 'StackerEnsembleModel_RF', 'WeightedEnsembleModel', 'StackerEnsembleModel_XT', 'StackerEnsembleModel_XGBoost', 'StackerEnsembleModel_CatBoost'}\n",
      "Bagging used: True  (with 8 folds)\n",
      "Multi-layer stack-ensembling used: True  (with 3 levels)\n",
      "Feature Metadata (Processed):\n",
      "(raw dtype, special dtypes):\n",
      "('float', []) : 17 | ['cat_ordered_base', 'cat_plain_base', 'lgb2_gbdt_base', 'lgb2_goss_base', 'lgb2_rf_base', ...]\n",
      "('int', [])   :  1 | ['Unnamed: 0']\n",
      "Plot summary of models saved to file: /kaggle/working/AutogluonModels/ag-20241204_105828SummaryOfModels.html\n",
      "*** End of fit() summary ***\n"
     ]
    }
   ],
   "source": [
    "predictor = TabularPredictor(label='Premium_Amount', \n",
    "                            eval_metric='rmse',\n",
    "                            ) \n",
    "\n",
    "TIME = 3600*5\n",
    "\n",
    "predictor.fit(train,\n",
    "              presets='best_quality',\n",
    "              time_limit=TIME,\n",
    "              verbosity=1,\n",
    "              excluded_model_types=['KNN'],\n",
    "              ag_args_fit={'num_gpus': 1}\n",
    "              )\n",
    "\n",
    "predictor.save(\"AutoGluon_Model\")\n",
    "results = predictor.fit_summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b87888dd",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-12-04T15:59:03.479619Z",
     "iopub.status.busy": "2024-12-04T15:59:03.478967Z",
     "iopub.status.idle": "2024-12-04T15:59:03.497983Z",
     "shell.execute_reply": "2024-12-04T15:59:03.497154Z"
    },
    "papermill": {
     "duration": 0.027657,
     "end_time": "2024-12-04T15:59:03.499543",
     "exception": false,
     "start_time": "2024-12-04T15:59:03.471886",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>score_val</th>\n",
       "      <th>eval_metric</th>\n",
       "      <th>pred_time_val</th>\n",
       "      <th>fit_time</th>\n",
       "      <th>pred_time_val_marginal</th>\n",
       "      <th>fit_time_marginal</th>\n",
       "      <th>stack_level</th>\n",
       "      <th>can_infer</th>\n",
       "      <th>fit_order</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WeightedEnsemble_L3</td>\n",
       "      <td>-1.045002</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>137.357594</td>\n",
       "      <td>13280.124841</td>\n",
       "      <td>0.015671</td>\n",
       "      <td>1.770248</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WeightedEnsemble_L2</td>\n",
       "      <td>-1.045053</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>101.808019</td>\n",
       "      <td>8417.333692</td>\n",
       "      <td>0.015976</td>\n",
       "      <td>1.487336</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LightGBMXT_BAG_L2</td>\n",
       "      <td>-1.045101</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>118.528643</td>\n",
       "      <td>9019.171870</td>\n",
       "      <td>7.015970</td>\n",
       "      <td>117.946934</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LightGBM_BAG_L2</td>\n",
       "      <td>-1.045168</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>115.705110</td>\n",
       "      <td>9004.995558</td>\n",
       "      <td>4.192436</td>\n",
       "      <td>103.770621</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LightGBM_BAG_L1</td>\n",
       "      <td>-1.045182</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>4.918388</td>\n",
       "      <td>92.356826</td>\n",
       "      <td>4.918388</td>\n",
       "      <td>92.356826</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>XGBoost_BAG_L1</td>\n",
       "      <td>-1.045183</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>1.528545</td>\n",
       "      <td>68.977578</td>\n",
       "      <td>1.528545</td>\n",
       "      <td>68.977578</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LightGBMXT_BAG_L1</td>\n",
       "      <td>-1.045196</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>17.215012</td>\n",
       "      <td>154.019336</td>\n",
       "      <td>17.215012</td>\n",
       "      <td>154.019336</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CatBoost_BAG_L1</td>\n",
       "      <td>-1.045244</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>0.274791</td>\n",
       "      <td>84.403559</td>\n",
       "      <td>0.274791</td>\n",
       "      <td>84.403559</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NeuralNetFastAI_BAG_L1</td>\n",
       "      <td>-1.045292</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>6.995614</td>\n",
       "      <td>1384.999068</td>\n",
       "      <td>6.995614</td>\n",
       "      <td>1384.999068</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ExtraTreesMSE_BAG_L1</td>\n",
       "      <td>-1.046529</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>28.833366</td>\n",
       "      <td>335.394234</td>\n",
       "      <td>28.833366</td>\n",
       "      <td>335.394234</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LightGBMLarge_BAG_L1</td>\n",
       "      <td>-1.046633</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>3.977910</td>\n",
       "      <td>82.766158</td>\n",
       "      <td>3.977910</td>\n",
       "      <td>82.766158</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>RandomForestMSE_BAG_L1</td>\n",
       "      <td>-1.047952</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>42.301118</td>\n",
       "      <td>6380.099314</td>\n",
       "      <td>42.301118</td>\n",
       "      <td>6380.099314</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>RandomForestMSE_BAG_L2</td>\n",
       "      <td>-1.049720</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>130.325953</td>\n",
       "      <td>13160.407660</td>\n",
       "      <td>18.813280</td>\n",
       "      <td>4259.182723</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NeuralNetTorch_BAG_L1</td>\n",
       "      <td>-1.072592</td>\n",
       "      <td>root_mean_squared_error</td>\n",
       "      <td>5.467930</td>\n",
       "      <td>318.208864</td>\n",
       "      <td>5.467930</td>\n",
       "      <td>318.208864</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     model  score_val              eval_metric  pred_time_val  \\\n",
       "0      WeightedEnsemble_L3  -1.045002  root_mean_squared_error     137.357594   \n",
       "1      WeightedEnsemble_L2  -1.045053  root_mean_squared_error     101.808019   \n",
       "2        LightGBMXT_BAG_L2  -1.045101  root_mean_squared_error     118.528643   \n",
       "3          LightGBM_BAG_L2  -1.045168  root_mean_squared_error     115.705110   \n",
       "4          LightGBM_BAG_L1  -1.045182  root_mean_squared_error       4.918388   \n",
       "5           XGBoost_BAG_L1  -1.045183  root_mean_squared_error       1.528545   \n",
       "6        LightGBMXT_BAG_L1  -1.045196  root_mean_squared_error      17.215012   \n",
       "7          CatBoost_BAG_L1  -1.045244  root_mean_squared_error       0.274791   \n",
       "8   NeuralNetFastAI_BAG_L1  -1.045292  root_mean_squared_error       6.995614   \n",
       "9     ExtraTreesMSE_BAG_L1  -1.046529  root_mean_squared_error      28.833366   \n",
       "10    LightGBMLarge_BAG_L1  -1.046633  root_mean_squared_error       3.977910   \n",
       "11  RandomForestMSE_BAG_L1  -1.047952  root_mean_squared_error      42.301118   \n",
       "12  RandomForestMSE_BAG_L2  -1.049720  root_mean_squared_error     130.325953   \n",
       "13   NeuralNetTorch_BAG_L1  -1.072592  root_mean_squared_error       5.467930   \n",
       "\n",
       "        fit_time  pred_time_val_marginal  fit_time_marginal  stack_level  \\\n",
       "0   13280.124841                0.015671           1.770248            3   \n",
       "1    8417.333692                0.015976           1.487336            2   \n",
       "2    9019.171870                7.015970         117.946934            2   \n",
       "3    9004.995558                4.192436         103.770621            2   \n",
       "4      92.356826                4.918388          92.356826            1   \n",
       "5      68.977578                1.528545          68.977578            1   \n",
       "6     154.019336               17.215012         154.019336            1   \n",
       "7      84.403559                0.274791          84.403559            1   \n",
       "8    1384.999068                6.995614        1384.999068            1   \n",
       "9     335.394234               28.833366         335.394234            1   \n",
       "10     82.766158                3.977910          82.766158            1   \n",
       "11   6380.099314               42.301118        6380.099314            1   \n",
       "12  13160.407660               18.813280        4259.182723            2   \n",
       "13    318.208864                5.467930         318.208864            1   \n",
       "\n",
       "    can_infer  fit_order  \n",
       "0        True         14  \n",
       "1        True         10  \n",
       "2        True         11  \n",
       "3        True         12  \n",
       "4        True          2  \n",
       "5        True          7  \n",
       "6        True          1  \n",
       "7        True          4  \n",
       "8        True          6  \n",
       "9        True          5  \n",
       "10       True          9  \n",
       "11       True          3  \n",
       "12       True         13  \n",
       "13       True          8  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "predictor.leaderboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "19d928e1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-12-04T15:59:03.514630Z",
     "iopub.status.busy": "2024-12-04T15:59:03.514081Z",
     "iopub.status.idle": "2024-12-04T16:02:10.158499Z",
     "shell.execute_reply": "2024-12-04T16:02:10.157608Z"
    },
    "papermill": {
     "duration": 186.65356,
     "end_time": "2024-12-04T16:02:10.160577",
     "exception": false,
     "start_time": "2024-12-04T15:59:03.507017",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n",
      "/opt/conda/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
      "  return torch.load(io.BytesIO(b))\n"
     ]
    }
   ],
   "source": [
    "y_pred = predictor.predict(test)\n",
    "y_pred = np.expm1(y_pred)\n",
    "\n",
    "sub = pd.read_csv('/kaggle/input/playground-series-s4e12/sample_submission.csv')\n",
    "sub['Premium Amount'] = y_pred\n",
    "sub.to_csv('submission.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kaggle": {
   "accelerator": "gpu",
   "dataSources": [
    {
     "databundleVersionId": 10305135,
     "sourceId": 84896,
     "sourceType": "competition"
    },
    {
     "datasetId": 6226700,
     "sourceId": 10098047,
     "sourceType": "datasetVersion"
    }
   ],
   "dockerImageVersionId": 30804,
   "isGpuEnabled": true,
   "isInternetEnabled": true,
   "language": "python",
   "sourceType": "notebook"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 18288.50141,
   "end_time": "2024-12-04T16:02:15.587059",
   "environment_variables": {},
   "exception": null,
   "input_path": "__notebook__.ipynb",
   "output_path": "__notebook__.ipynb",
   "parameters": {},
   "start_time": "2024-12-04T10:57:27.085649",
   "version": "2.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
